src/ica_api.c

```diff
@@ -361,12 +361,6 @@ unsigned int ica_open_adapter(ica_adapter_handle_t *adapter_handle)
 		if (*adapter_handle == -1)
 			*adapter_handle = open(DEFAULT3_CRYPT_DEVICE, O_RDWR);
 	}
-	if (*adapter_handle != -1) {
```

Comment:
While you are at it:
The statement `*adapter_handle = DRIVER_NOT_LOADED;` does not make much sense. All code paths below will set `*adapter_handle`, so that assignment is superfluous. 
Nevertheless, `*adapter_handle` can be left at -1 (if the very last open fails), and -1 happens to be `DRIVER_NOT_LOADED`....